File: HRMS/HRMS.Web/Modules/ServerTypes/Recruitment/InterviewsForm.ts
import { DateEditor, EnumEditor, initFormType, IntegerEditor, LookupEditor, PrefixedContext, TextAreaEditor } from "@serenity-is/corelib";
import { InterviewRound } from "./InterviewRound";

export interface InterviewsForm {
    CandidateId: LookupEditor;
    InterviewerId: LookupEditor;
    InterviewDate: DateEditor;
    Round: EnumEditor;
    Rating: IntegerEditor;
    Comments: TextAreaEditor;
}

export class InterviewsForm extends PrefixedContext {
    static readonly formKey = 'Recruitment.Interviews';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!InterviewsForm.init) {
            InterviewsForm.init = true;

            var w0 = LookupEditor;
            var w1 = DateEditor;
            var w2 = EnumEditor;
            var w3 = IntegerEditor;
            var w4 = TextAreaEditor;

            initFormType(InterviewsForm, [
                'CandidateId', w0,
                'InterviewerId', w0,
                'InterviewDate', w1,
                'Round', w2,
                'Rating', w3,
                'Comments', w4
            ]);
        }
    }
}

[InterviewRound]; // referenced types

File: HRMS/HRMS.Web/Modules/Recruitment/Interviews/InterviewsForm.cs
using Serenity.ComponentModel;
using HRMS.HR;

namespace HRMS.Recruitment.Forms;

[FormScript("Recruitment.Interviews")]
[BasedOnRow(typeof(InterviewsRow), CheckNames = true)]
public class InterviewsForm
{
    [LookupEditor(typeof(CandidatesRow))]
    public int CandidateId { get; set; }
    [LookupEditor(typeof(EmployeeRow))]
    public int InterviewerId { get; set; }
    [DefaultValue("now")]
    public DateTime InterviewDate { get; set; }
    public int Round { get; set; }
    public int Rating { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Comments { get; set; }
    [DisplayName("Interview Done")]
    public bool IsCompleted { get; set; }
}
